src/map.cpp

```diff
@@ -526,13 +533,20 @@ bool checkSteepLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t
 	return true;
 }
 
-bool checkSlightLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t z)
+bool checkSlightLine(uint16_t x0, uint16_t y0, uint16_t x1, uint16_t y1, uint8_t z, float modifier = 0)
 {
 	float dx = x1 - x0;
-	float slope = (dx == 0) ? 1 : (y1 - y0) / dx;
+	float slope = (dx == 0) ? 1 : std::abs((y1 - y0) / dx);
+	if (y1 < y0) {
+		slope *= -1;
+	}
 	float yi = y0 + slope;
+	if (modifier != 0) {
+		yi += modifier;
+	}
 
 	for (uint16_t x = x0 + 1; x < x1; ++x) {
+		// g_game.addMagicEffect(Position(x,std::floor(yi + 0.1),z), CONST_ME_EXPLOSIONHIT); // debugging
```

Comment:
What do these codes commented in a PR do to merge?

Reply:
Debug logic. It can be removed, at the small hindsight of TFS code pollution I propose that it shall have precompiler variables to debug certain scenarios, this being one of them.